src/main/resources/lib/ssb/utils/solrUtils.ts

```diff
@@ -0,0 +1,181 @@
+import { HttpResponse } from 'enonic-types/http'
+
+const SOLR_PARAM_QUERY: string = 'q'
+const SOLR_FORMAT: string = 'json'
+const SOLR_BASE_URL: string = app.config && app.config['ssb.solrFriTekstSok.baseUrl'] ? app.config['ssb.solrFriTekstSok.baseUrl'] :
+  'https://i.ssb.no/solrmaster/fritekstsok/select'
+
+
+const {
+  request
+} = __non_webpack_require__('/lib/http-client')
+
+const {
+  moment
+} = __non_webpack_require__('/lib/vendor/moment')
+
+
+export function solrSearch(term: string, language: string, numberOfHits: number, start: number = 0): SolrPrepResultAndTotal {
+  const searchResult: SolrResult | undefined = querySolr({
+    query: createQuery(term, language, numberOfHits, start)
+  })
+  return searchResult ? {
+    hits: nerfSearchResult(searchResult, language),
+    total: searchResult.grouped.gruppering.matches
+  } : {
+    hits: [],
+    total: 0
+  }
+}
+
+
+function nerfSearchResult(solrResult: SolrResult, language: string): Array<PreparedSearchResult> {
+  return solrResult.grouped.gruppering.groups.reduce((acc: Array<PreparedSearchResult>, group) => {
+    group.doclist.docs.forEach((doc: SolrDoc) => {
+      const highlight: SolrHighlighting | undefined = solrResult.highlighting[doc.id]
+      acc.push({
+        title: highlight.tittel ? highlight.tittel[0] : doc.tittel,
+        preface: highlight.innhold ? highlight.innhold[0] : doc.tittel,
+        contentType: doc.innholdstype,
+        url: doc.url,
+        mainSubject: doc.hovedemner.split(';')[0],
+        publishDate: doc.publiseringsdato,
+        publishDateHuman: doc.publiseringsdato ? moment(doc.publiseringsdato).locale(language).format('Do MMMM YYYY') : ''
```

Comment:
This might have to be formatted differently for english view? 
Currently, the format being used for e.g. statistics is 'LL' and language has to return 'en-gb' if language is en. 
The norwegian date should still show correctly. Though there have been reports about the dates (for all languages) being buggy after the moment changes.

Reply:
Changed to use 'en-gb' or 'nb-no', and LL now.

Reply:
It should suffice with just nb instead of nb-no!